File: 이것이_취업을_위한_코딩_테스트다/ch4 Implementation/Clock.java
import java.util.Scanner;

public class Clock {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        int cnt = 0;
        for (int h = 0; h <= n; h++) {
            for (int m = 0; m < 60; m++) {
                for (int s = 0; s < 60; s++) {
                    // 둘째자리가 3이면
                    if ((h / 10) == 3 || (m / 10) == 3 || (s / 10) == 3) {
                        cnt++;
                    }
                    // 첫째자리가 3이면
                    else if ((h % 10) == 3 || (m % 10) == 3 || (s % 10) == 3) {
                        cnt++;
                    }
                }
            }
        }

        System.out.println(cnt);

    }
}

File: 이것이_취업을_위한_코딩_테스트다/ch4 Implementation/clock.py
# 시각 입력 받기
n = int(input())
cnt = 0

# 0~n까지
for h in range(n + 1):
    # 0~59까지
    for m in range(60):
        # 0~59까지
        for s in range(60):
            # 문자열로 바꾸고 문자열에 3이 있으면
            if "3" in str(h) + str(m) + str(s):
                cnt += 1

print(cnt)
